spring-kafka/src/main/java/org/springframework/kafka/support/converter/BatchMessagingMessageConverter.java

```diff
@@ -60,12 +60,14 @@
  * @author Gary Russell
  * @author Dariusz Szablinski
  * @author Biju Kunjummen
+ * @author Sanghyeok An
  * @since 1.1
  */
 public class BatchMessagingMessageConverter implements BatchMessageConverter {
 
 	protected final LogAccessor logger = new LogAccessor(LogFactory.getLog(getClass())); // NOSONAR
 
+	@Nullable
```

Comment:
`RecordMessageConverter` can be null.  Please refer to codes below.

```java
// BatchMessagingMessageConverter.java
	public BatchMessagingMessageConverter() {
		this(null);
	}

	public BatchMessagingMessageConverter(RecordMessageConverter recordConverter) {
		this.recordConverter = recordConverter;
		if (JacksonPresent.isJackson2Present()) {
			this.headerMapper = new DefaultKafkaHeaderMapper();
		}
	}
```

Then, `getRecordMessageConverter()` can return `null`.
`BatchMessagingMessageListenerAdapter#setBatchMessageConverter()` call `getRecordMessageConverter()` internally to set messageConverter. 

If user created `BatchMessagingMessageConverter` by using default constructor, there is no message converter even if `BatchMessagingMessageListenerAdapter#setBatchMessageConverter()` is completed without any warning logs.